rotkehlchen/inquirer.py

```diff
@@ -192,37 +263,40 @@ def __new__(
 
         return Inquirer.__instance
 
+    @staticmethod
+    def get_oracle_instances(
+        oracle_order: Sequence[CurrentPriceOracle],
+        cryptocompare: 'Cryptocompare',
+        coingecko: 'Coingecko',
+    ) -> List[Tuple[CurrentPriceOracle, CurrentPriceOracleInstance]]:
+        oracle_instances: List[Tuple[CurrentPriceOracle, CurrentPriceOracleInstance]]
+        oracle_instances = []
+        for oracle in oracle_order:
+            if oracle == CurrentPriceOracle.COINGECKO:
+                oracle_instances.append((oracle, coingecko))
+            elif oracle == CurrentPriceOracle.CRYPTOCOMPARE:
+                oracle_instances.append((oracle, cryptocompare))
+            else:
+                raise AssertionError(f'Unexpected CurrentPriceOracle: {oracle}')
+
+        return oracle_instances
+
     @staticmethod
     def inject_ethereum(ethereum: 'EthereumManager') -> None:
         Inquirer()._ethereum = ethereum
 
     @staticmethod
-    def _cryptocompare_find_usd_price(asset: Asset) -> Price:
-        """Returns the current USD price of the asset
-
-        Return Price(ZERO) if price can't be found.
-
-        May raise:
-        - RemoteError if the cryptocompare query has a problem
-        """
-        try:
-            result = Inquirer()._cryptocompare.query_endpoint_price(
-                from_asset=asset,
-                to_asset=A_USD,
-            )
-        except PriceQueryUnsupportedAsset:
-            log.error(
-                'Cryptocompare usd price for asset failed since it is not known to cryptocompare',
-                asset=asset,
-            )
-            return Price(ZERO)
-        if 'USD' not in result:
-            log.error('Cryptocompare usd price query failed', asset=asset)
-            return Price(ZERO)
-
-        price = Price(FVal(result['USD']))
-        log.debug('Got usd price from cryptocompare', asset=asset, price=price)
-        return price
+    def set_oracle_order(oracle_order: Sequence[CurrentPriceOracle]) -> None:
+        if set(oracle_order) != set(CurrentPriceOracle):
+            raise AssertionError('All current price oracles are required')
+
+        oracle_instances = Inquirer().get_oracle_instances(
+            oracle_order=oracle_order,
+            cryptocompare=Inquirer()._cryptocompare,
+            coingecko=Inquirer()._coingecko,
+        )
+        Inquirer()._oracle_order = oracle_order
+        Inquirer()._oracle_instances = oracle_instances
 
     @staticmethod
     def find_usd_price(asset: Asset, ignore_cache: bool = False) -> Price:
```

Comment:
I've just realised that preventing the extra queries on the new current price endpoint is gonna be much easier after merging this PR (again. if the proposed solution here is good).

Now that `find_usd_price` calls:
```python
price = oracle_instance.query_current_price(
    from_asset=asset,
    to_asset=A_USD,
 )
```

What I will do either in another PR or in this one:
1. Rename `find_usd_price` as `find_price` enabling `to_asset` argument. Also remove 'USD' from its logs and use `to_asset.identifier`.
2. Amend the cache for storing tuples as key: `(from_asset, to_asset)`.
3. Create a new `find_usd_price` that will call `find_price(from_asset=asset, to_asset=A_USD, ignore_cache=ignore_cache)`